frontend/src/components/TreeViewer/TreeViewer.jsx

```diff
@@ -1,11 +1,79 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import styles from "./TreeViewer.module.css";
-import { useParams } from "react-router-dom";
 import ReactJson from "react-json-view";
+import { Tree } from "../Tree/Tree";
+
+const MAX_DEPTH = 2;
+
+/**
+ * Given the original data which has an unbounded number
+ * of levels, this function trims the total depth of the tree
+ * to the given depth.
+ */
+function extractObjectWithMaxDepth(obj, depth = MAX_DEPTH) {
+  if (depth < 0) {
+    return null;
+  }
+
+  // Recursively add children of the object if they do not
+  // exceed the max depth
+  return {
+    name: obj.name,
+    children: obj.children
+      ? obj.children
+          .map((node) => extractObjectWithMaxDepth(node, depth - 1))
+          .filter((node) => node !== null)
+      : [],
+  };
+}
+
+/**
+ * Creates an index so that we can get the reference to
+ * a particular node, given its name, in constant time.
+ *
+ * TODO: Confirm if duplicate names are possible or not
```

Comment:
Let's ask the team this either in chat or in meeting

Reply:
Is a node rendered with 0 children considered a depth of 1?

Reply:
The root node with 0 children has a depth of 0 actually

Reply:
Then why do we have MAX_DEPTH=2 ? Aren't we supposed to show only the node's children?

Reply:
Duplicate names are not allowed in the tree, so this mapping function should work

Reply:
> Then why do we have MAX_DEPTH=2 ? Aren't we supposed to show only the node's children?

I just put 2 for now since there's enough space for the content to fit properly, even with 2 levels of children, but we can change that anytime.